nab_evaluation.py

```diff
@@ -0,0 +1,144 @@
+"""
+    Evaluation module for point-wise prediction algorithms in terms of NAB (Numenta Anomaly Benchmark) score.
+"""
+import numpy as np
+
+def scaledSigmoid(relativePositionInWindow):
+    if relativePositionInWindow > 3.0:
+        # FP well behind window
+        return -1.0
+    else:
+        return 2/(np.exp(5*relativePositionInWindow) + 1) - 1.0
+
+
+def label_anomaly_windows(labels):
+    """
+    Converts the point wise anomaly labels to windows
+    :param label (type: 1D array): true labels 1/0 for each point
+    :return (type: list): set of intervals of anomaly (start time, end time) sorted in increasing order of start time
+    """
+   
+    labeled_anomaly_window = []
+    #if the anomaly window starts from the beginning
+    start = 0
+    for i in range(1, labels.shape[0]):
+        if labels[i] == 1 and labels[i-1] == 0:
+            start = i
+        elif labels[i-1] == 1 and labels[i] == 0:
+            end = i-1
+            labeled_anomaly_window.append((start, end))
+        elif i == len(labels)-1 and labels[i] == 1:
+            #if the anomaly window extends till the end
+            labeled_anomaly_window.append((start, i))
+              
+    return labeled_anomaly_window
+
+
+def getCorrespondingWindow(index, windows):
+    """
+    Finds the corresponding window to the each predicted anomaly
+    :param index (type:int): index of predicted anomaly point
+    :param windows (type list of tuples (start, end)): list of true anomaly windows
+    :return (typ: tuple): anomaly window if point lies inside it else the preceding window
+    """
+    
+    for i, window in enumerate(windows):
+        if index <= window[1] and index >= window[0]:
```

Comment:
`window[0] <= index and index <= window[1]` is easier to read.

Reply:
Done.